cloner/default.go

```diff
@@ -66,9 +68,25 @@ func (c *cloner) Clone(ctx context.Context, params Params) error {
 	}
 	// clone the repository
 	r, err := git.PlainClone(params.Dir, false, opts)
-	if err != nil {
+	if errors.Is(plumbing.ErrReferenceNotFound, err) && !strings.HasPrefix(params.Ref, "refs/") {
+		// If params.Ref is provided without refs/*, then we are assuming it to either refs/heads/ or refs/tags.
+		// Try clone again with inverse ref.
+		if opts.ReferenceName.IsBranch() {
+			opts.ReferenceName = plumbing.ReferenceName("refs/tags/" + params.Ref)
```

Comment:
Can you confirm if cloning with a tag and depth set works? Because the depth is being set on L61.
Also where is `expandRef` method defined (used in L56). We should just augment that function to expand the ReferenceName based on the conditions (IsBranch and IsTag).

Reply:
Cloning with tag and depth works. expandRef is defined in util.go file. 
I can create an inverseRef method that returns inverse but it won't be intuitive. expandRef has a specific utility. We won't be able to add to it.

Reply:
Ack.